src/js/plugins/plugin.resume.js

```diff
@@ -0,0 +1,88 @@
+/* global BookReader */
+
+/**
+ * Plugin to remember the current page number in a cookie
+ */
+jQuery.extend(BookReader.defaultOptions, {
+  enablePageResume: true,
+  /**
+   * @const null|string eg '/', '/details/id'
+   */
+  resumeCookiePath: null,
+});
+
+/**
+ * @override BookReader.init
+ */
+BookReader.prototype.init = (function(super_) {
+  return function() {
+    super_.call(this);
+    if (this.options.enablePageResume) {
+      this.bind(BookReader.eventNames.fragmentChange, function() {
+        const params = this.paramsFromCurrent();
+        this.updateResumeValue(params.index);
+      }.bind(this));
+    }
+  };
+})(BookReader.prototype.init);
+
+/**
+ * Get's the page resume value, for remembering reader's page
+ * Can be overriden for different implementation
+ *
+ * @return {Number|null}
+ */
+BookReader.prototype.getResumeValue = function() {
+  const val = BookReader.docCookies.getItem('br-resume');
+  if (val !== null) return parseInt(val);
+  else return null;
+}
+
+/**
+ * Set's the page resume value, for remembering reader's page
+ * Can be overriden for different implementation
+ *
+ * @param {Number} index leaf index
+ * @param {String} cookieName
+ */
+BookReader.prototype.updateResumeValue = function(index, cookieName) {
+  const ttl = new Date(+new Date + 12096e5); // 2 weeks
+  const path = this.options.rsumeCookiePath || window.location.pathname;
+  cookieName = cookieName || 'br-resume';
+  BookReader.docCookies.setItem(cookieName, index, ttl, path, null, false);
+}
+
+/**
+ * Use to get, set and remove item from cookie
+ *
+ * See more:
+ *  https://developer.mozilla.org/en-US/docs/Web/API/document.cookie
+ *  https://developer.mozilla.org/User:fusionchess
+ *  https://github.com/madmurphy/cookies.js
+ *  This framework is released under the GNU Public License, version 3 or later.
+ *  http://www.gnu.org/licenses/gpl-3.0-standalone.html
+ */
+BookReader.docCookies = {
```

Comment:
This looks like a good candidate for pulling out in a separate file; but make sure it's still exposed here! So something like:

plugins/plugin.resume.js:
```js
import * as docCookies from '../utils/docCookies.js';
...
/** @deprecated Exposed for backward compatibility */
BookReader.docCookies = docCookies;
```

utils/docCookies.js
```js
export function getItem() { ... }
export function setItem() { ... }
export function hasItem() { ... }
```